prism/src/main/java/org/apache/falcon/resource/proxy/SchedulableEntityManagerProxy.java

```diff
@@ -378,6 +385,32 @@ protected APIResult doExecute(String colo) throws FalconException {
         return consolidateResult(results, APIResult.class);
     }
 
+    private void isEntityPartOfAnExtension(Entity entity) {
+        String tags = entity.getTags();
+        checkExtensionJobExist(tags);
+    }
+
+    private void checkExtensionJobExist(String tags) {
+        if (tags != null) {
+            String jobName = ExtensionManager.getJobNameFromTag(tags);
+            ExtensionMetaStore extensionMetaStore = ExtensionStore.getMetaStore();
+            if (jobName != null && extensionMetaStore.checkIfExtensionJobExists(jobName)) {
+                throw FalconWebException.newAPIException("Entity operation is not allowed on this entity as it is"
+                        + "part of an extension job:" + jobName);
+            }
+        }
+    }
+
+    private void isEntityPartOfAnExtension(String type, String entityName) {
+        String tags;
+        try {
+            tags = EntityUtil.getEntity(type, entityName).getTags();
```

Comment:
You can just do a isEntityPartOfAnExtension(EntityUtil.getEntity(type, entityName))